docs/reference/contributing/storage/BlockDevice.md

```diff
@@ -12,20 +16,35 @@
 
 #### Defined behavior
 
-[Include any defined behavior in bullet format here.]
+* Erase leaves memory as undefined, it does not set memory to a predetermined value.
 
 #### Undefined behavior
 
-[Include any undefined behavior in bullet format here.]
+* Programming without erase is undefined behaviour don't do it silly billy.
+
+#### Notes
 
-#### Potential bugs
+![blockdevicesectors](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/blockdevice_block_size.png)
 
-[Include any potential bugs in bullet format here.]
+Erase, Program, and Read block sizes my not be the same; however, they must be multiples of each other.
 
 ### Implementing BlockDevice
 
-[Include implementation information here.]
+You can find the BlockDevice class on the master branch under the features/filesystem/bd path in Mbed OS.
+
+[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/class_block_device.html)
+
+The primary functions to implement are:
+* `int read(void *buffer, bd_addr_t addr, bd_size_t size);`
+* `int program(void *buffer, bd_addr_t addr, bd_size_t size);`
+* `int erase(bd_addr_t addr, bd_size_t size);`
 
 ### Testing
 
-[Include testing information here.]
+You can run blockdevice tests for heap, mbr, and util block devices with the following command.
+
+```
+mbed test -t <toolchain> -m <target> -n features-tests-filesystem-*_block_device
+```
+
+The simpliest way to add tests for new block devices is to copy an existing implementation such as HeapBlocKDevice <-- TODO: Linky linky? --> and changing the block device class to your own.
```

Comment:
Github links are a no go right? If I can't link directly to the test is there somewhere else the tests get uploaded to?

Reply:
Tests should be in the Doxygen. You should be able to find them somewhere here: https://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/annotated.html

[It might be in modules if it's not in data structures.]